aiida/orm/implementation/general/calculation/job/__init__.py

```diff
@@ -118,6 +118,28 @@ def process(cls):
     def get_builder(cls):
         return cls.process().get_builder()
 
+    def get_builder_restart(self):
+        from aiida.work.job_processes import JobProcess
+        from aiida.work.ports import PortNamespace
+
+        inputs = self.get_inputs_dict()
+        options = self.get_options()
+        builder = self.get_builder()
+
+        for port_name, port in self.process().spec().inputs.items():
+            if port_name == JobProcess.OPTIONS_INPUT_LABEL:
+                setattr(builder, port_name, options)
+            elif isinstance(port, PortNamespace):
+                namespace = port_name + '_'
+                sub = {link.replace(namespace, ''): node for link, node in inputs.items() if link.startswith(namespace)}
```

Comment:
I'm not sure `link.replace(namespace, '')` is safe, e.g. if the sub-namespace has the same name it would remove both. Better `link[len(namespace):]` or something equivalent

Reply:
This is fixed